connect/proxy/proxy.go

```diff
@@ -3,19 +3,23 @@ package proxy
 import (
 	"crypto/x509"
 
+	"github.com/hashicorp/consul/lib/telemetry"
+
+	"github.com/armon/go-metrics"
+
 	"github.com/hashicorp/consul/api"
 	"github.com/hashicorp/consul/connect"
-	"github.com/hashicorp/consul/lib"
 	"github.com/hashicorp/go-hclog"
 )
 
 // Proxy implements the built-in connect proxy.
 type Proxy struct {
-	client     *api.Client
-	cfgWatcher ConfigWatcher
-	stopChan   chan struct{}
-	logger     hclog.Logger
-	service    *connect.Service
+	client        *api.Client
+	cfgWatcher    ConfigWatcher
+	stopChan      chan struct{}
+	logger        hclog.Logger
+	service       *connect.Service
+	metricsClient *metrics.Metrics
```

Comment:
Should this be an interface instead of the concrete type?

Reply:
Yep! Just a little placeholder to make sure the draft compiled and it would break once I started changing the init returns!